src/logging.c

```diff
@@ -78,33 +87,40 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
       g_rcutils_logging_output_format_string[chars_to_copy] = '\0';
     } else {
       if (NULL != ret_str) {
-        fprintf(
-          stderr,
-          "Failed to get output format from env. variable: %s. Using default output format.\n",
-          ret_str);
-        ret = RCUTILS_RET_INVALID_ARGUMENT;
+        if (rcutils_error_is_set()) {
+          fprintf(stderr, "Overwriting error message: %s\n", rcutils_get_error_string_safe());
+          rcutils_reset_error();
+        }
+        RCUTILS_SET_ERROR_MSG(
+          "Failed to get output format from env. variable. Using default output format.",
+          g_rcutils_logging_allocator);
+        if (ret != RCUTILS_RET_OK) {
+          ret = RCUTILS_RET_MULTIPLE_ERRORS;
+        } else {
+          ret = RCUTILS_RET_INVALID_ARGUMENT;
+        }
       }
       memcpy(g_rcutils_logging_output_format_string, g_rcutils_logging_default_output_format,
         strlen(g_rcutils_logging_default_output_format) + 1);
     }
 
-    if (!rcutils_allocator_is_valid(&allocator)) {
-      fprintf(
-        stderr,
-        "Provided allocator is invalid. Using the default allocator.\n");
-      ret = RCUTILS_RET_INVALID_ARGUMENT;
-      allocator = rcutils_get_default_allocator();
-    }
-    g_rcutils_logging_allocator = allocator;
     g_rcutils_logging_severities_map = rcutils_get_zero_initialized_string_map();
     rcutils_ret_t string_map_ret = rcutils_string_map_init(
       &g_rcutils_logging_severities_map, 0, g_rcutils_logging_allocator);
     if (string_map_ret != RCUTILS_RET_OK) {
       fprintf(
         stderr,
-        "Failed to initialize map for logger severities. Severities will not be configurable.\n");
+        "Failed to initialize logging severities map: %s\n", rcutils_get_error_string_safe());
+      rcutils_reset_error();
```

Comment:
I would incorporate this into your new error state but not print anything here. Consider it being something like this:

```python
try:
  init_map(...)
except SomeMapError as exc:
  raise MyError("could not setup logging, because: %s" % str(exc))
```

Basically you're catching the string map error, and "re throwing" you're own more applicable error.

Reply:
10373ef429e41951f09ad594b495c750a2ec1b76 does this :+1: